testdata/requires/Taskfile.yml

```
@@ -10,6 +10,14 @@ tasks:
         - foo
     cmd: echo "{{.foo}}"
 
+  var-defined-in-task:
+    vars:
+      foo: bar
+    requires:
+      vars:
+        - foo
+    cmd: echo "{{.foo}}"
```

Just to follow our style convention:

```suggestion
  var-defined-in-task:
    vars:
      FOO: bar
    requires:
      vars:
        - FOO
    cmd: echo "{{.FOO}}"
```

---

You're right!